test/e2e/framework/pv/pv.go

```diff
@@ -844,3 +848,22 @@ func WaitForPersistentVolumeDeleted(c clientset.Interface, pvName string, poll,
 	}
 	return fmt.Errorf("PersistentVolume %s still exists within %v", pvName, timeout)
 }
+
+// WaitForFinalizerAdded waits for a finalizer to be added to a PVC in
+// a given namespace.
+func WaitForFinalizerAdded(ctx context.Context, cs clientset.Interface, name, namespace, finalizer string, poll, timeout time.Duration) error {
+	defer ginkgo.GinkgoRecover()
+	if successful := utils.WaitUntil(poll, timeout, func() bool {
+		pvc, err := cs.CoreV1().PersistentVolumeClaims(namespace).Get(ctx, name, metav1.GetOptions{})
+		framework.ExpectNoError(err, fmt.Sprintf("failed to get pvc %s/%s", namespace, name))
+		for _, f := range pvc.Finalizers {
+			if f == finalizer {
+				return true
+			}
+		}
+		return false
+	}); successful {
+		return nil
+	}
+	return fmt.Errorf("finalizer not added to pvc %s/%s", namespace, name)
```

Comment:
Can you print out the finalizer in the err msg

Reply:
done